test/e2e/autoscaling/cluster_size_autoscaling.go

```diff
@@ -999,6 +1001,59 @@ var _ = SIGDescribe("Cluster size autoscaling [Slow]", func() {
+		schedulerName, found := framework.TestContext.ExtraParams[nonExistingIgnoredSchedulerNameKey]
+		if !found {
+			framework.Logf("Skipping test, Didn't find an ignored non-existent scheduler name to use")
+			return
+		}
+		// 70% of allocatable memory of a single node * replica count, forcing a scale up in case of normal pods
+		replicaCount := 2 * nodeCount
+		reservedMemory := int(float64(replicaCount) * float64(0.7) * float64(memAllocatableMb))
+		cleanupFunc := ReserveMemoryWithSchedulerName(ctx, f, "memory-reservation", replicaCount, reservedMemory, false, 1, schedulerName)
+		defer cleanupFunc()
+		// Verify that cluster size is increased
+		ginkgo.By("Waiting for cluster scale-up")
+		sizeFunc := func(size int) bool {
+			// Softly checks scale-up since other types of machines can be added which would affect #nodes
+			return size > nodeCount
+		}
+		framework.ExpectNoError(WaitForClusterSizeFuncWithUnready(ctx, f.ClientSet, sizeFunc, scaleUpTimeout, 0))
+	})
+	ginkgo.It("shouldn't scale up when unprocessed pod is created and is going to be schedulable[Feature:ClusterScaleUpBypassScheduler]", func(ctx context.Context) {
+		schedulerName, found := framework.TestContext.ExtraParams[nonExistingIgnoredSchedulerNameKey]
+		if !found {
+			framework.Logf("Skipping test, Didn't find an ignored non-existent scheduler name to use")
+			return
+		}
+		// 50% of allocatable memory of a single node, so that no scale up would trigger in normal cases
+		replicaCount := 1
+		reservedMemory := int(float64(0.5) * float64(memAllocatableMb))
+		cleanupFunc := ReserveMemoryWithSchedulerName(ctx, f, "memory-reservation", replicaCount, reservedMemory, false, 1, schedulerName)
+		defer cleanupFunc()
+		// Verify that cluster size is the same
+		ginkgo.By(fmt.Sprintf("Waiting for scale up hoping it won't happen, sleep for %s", scaleUpTimeout.String()))
+		time.Sleep(scaleUpTimeout)
+		sizeFunc := func(size int) bool {
+			return size == nodeCount
+		}
+		framework.ExpectNoError(WaitForClusterSizeFuncWithUnready(ctx, f.ClientSet, sizeFunc, time.Second, 0))
+	})
+	ginkgo.It("shouldn't scale up when unprocessed pod is created and scheduler is not specified to be bypassed[Feature:ClusterScaleUpBypassScheduler]", func(ctx context.Context) {
+		// 70% of allocatable memory of a single node * replica count, forcing a scale up in case of normal pods
+		replicaCount := 2 * nodeCount
+		reservedMemory := int(float64(replicaCount) * float64(0.7) * float64(memAllocatableMb))
+		schedulerName := "non-existent-scheduler-" + f.UniqueName
+		cleanupFunc := ReserveMemoryWithSchedulerName(ctx, f, "memory-reservation", replicaCount, reservedMemory, false, 1, schedulerName)
+		defer cleanupFunc()
+		// Verify that cluster size is the same
+		ginkgo.By(fmt.Sprintf("Waiting for scale up hoping it won't happen, sleep for %s", scaleUpTimeout.String()))
```

Comment:
IIUC we are not actually sleeping here, while we should.

Reply:
Yes, mb I forgot to keep it consistent with the prev test, Should be alright now